sopel/modules/remind.py

```diff
@@ -353,6 +353,33 @@ def __eq__(self, other):
             ]
         )
 
+    def __ne__(self, other):
+        return any(
+            getattr(self, attr) != getattr(other, attr, None)
+            for attr in [
+                'hour',
+                'minute',
+                'second',
+                'timezone',
+                'year',
+                'month',
+                'day',
+                'message',
+            ]
+        )
+
+    def __hash__(self):
```

Comment:
I've seen this warning, and I did some research on `__hash__` because I remembered something very important about it. From the [python documentation about `__hash__`](https://docs.python.org/3/reference/datamodel.html#object.__hash__) (emphasis is mine):

> If a class defines mutable objects and implements an `__eq__()` method, it **should not** implement `__hash__()`, since the implementation of hashable collections requires that a key’s hash value is immutable (if the object’s hash value changes, it will be in the wrong hash bucket).

Since you can change `TimeReminder`'s attributes, you **should not** implement the `__hash__` magic method.

Reply:
LGTM probably doesn't care whether the method is "implemented" or just set to `None`, indicating the object isn't hashable. I'd like to give them a few days (since it's the weekend) for looking into semmle/ql#2361 though, before I come back and tweak this.